go/store/blobstore/local.go

```diff
@@ -232,3 +211,22 @@ func (bs *LocalBlobstore) Exists(ctx context.Context, key string) (bool, error)
 
 	return err == nil, err
 }
+
+func (bs *LocalBlobstore) Contatenate(ctx context.Context, key string, sources []string) (ver string, err error) {
+	readers := make([]io.Reader, len(sources))
+	for i := range readers {
+		path := filepath.Join(bs.RootDir, sources[i]) + bsExt
+		if readers[i], err = os.Open(path); err != nil {
+			return "", err
```

Comment:
Nit: Is there a way to restructure this slightly so we get to Close() the opened readers in the error case without relying on in gc finalization?